cuda/include/vecmem/memory/cuda/resources/resources.hpp

```diff
@@ -0,0 +1,9 @@
+#include <memory_resource>
+
+#include "vecmem/memory/resources/base_resource.hpp"
+
+namespace vecmem::memory::resources {
+    base_resource * get_terminal_cuda_device_resource() noexcept;
+    base_resource * get_terminal_cuda_host_resource() noexcept;
+    base_resource * get_terminal_cuda_managed_resource() noexcept;
```

Comment:
I don't like this way of providing the resources to the users. :frowning:

The resources, eventually, will need to be tied to particular devices. So that in a system that has multiple GPUs, we could specify which resource handles which of them.

So I would just forego any centrally provided resources, and make it a task for the final application to create the resource objects for the GPUs / other resources that they want to use.

Reply:
That's a fair comment. This design mirrors roughly what the standard library does [(`std::pmr::get_default_resource`)](https://en.cppreference.com/w/cpp/memory/get_default_resource), but that doesn't mean it's the best design. Putting the onus of creating the resources on the user is fine with me.